schemavalidator/conflicts_with.go

```diff
@@ -0,0 +1,73 @@
+package schemavalidator
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-framework-validators/helpers/pathutils"
+	"github.com/hashicorp/terraform-plugin-framework-validators/helpers/validatordiag"
+	"github.com/hashicorp/terraform-plugin-framework/attr"
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/tfsdk"
+)
+
+// conflictsWithAttributeValidator is the underlying struct implementing ConflictsWith.
+type conflictsWithAttributeValidator struct {
+	pathExpressions path.Expressions
+}
+
+// ConflictsWith checks that a set of path.Expression,
+// including the attribute it's applied to, are not set simultaneously.
+// This implements the validation logic declaratively within the tfsdk.Schema.
+//
+// Relative path.Expression will be resolved against the validated attribute.
+func ConflictsWith(attributePaths ...path.Expression) tfsdk.AttributeValidator {
+	return &conflictsWithAttributeValidator{attributePaths}
+}
+
+var _ tfsdk.AttributeValidator = (*conflictsWithAttributeValidator)(nil)
+
+func (av conflictsWithAttributeValidator) Description(ctx context.Context) string {
+	return av.MarkdownDescription(ctx)
+}
+
+func (av conflictsWithAttributeValidator) MarkdownDescription(_ context.Context) string {
+	return fmt.Sprintf("Ensure that if an attribute is set, these are not set: %q", av.pathExpressions)
+}
+
+func (av conflictsWithAttributeValidator) Validate(ctx context.Context, req tfsdk.ValidateAttributeRequest, res *tfsdk.ValidateAttributeResponse) {
+	var v attr.Value
+	res.Diagnostics.Append(tfsdk.ValueAs(ctx, req.AttributeConfig, &v)...)
+	if res.Diagnostics.HasError() {
+		return
+	}
+
```

Comment:
Is this necessary? [`tfsdk.ValidateAttributeRequest.AttributeConfig` is already an `attr.Value`](https://pkg.go.dev/github.com/hashicorp/terraform-plugin-framework@v0.9.0/tfsdk#ValidateAttributeRequest.AttributeConfig)

Reply:
This might be an heritage of something that made sense when this PR first was created.

Indeed seems pointless and I should have spotted it. Will refactor.

Reply:
Yeah, this is as I suspected something I wrote when we had no `.IsNull() / .IsUnknown()` on the `attr.Value` interface. I guess I started renaming stuff and I missed this step was unnecessary now. 

Funny, that this might have been the very reason I proposed to add those methods 🤦